s_tui/s_tui.py

```diff
@@ -198,50 +205,63 @@ class GraphView(urwid.WidgetPlaceholder):
     The change is state should be reflected in the GraphController
     """
     def __init__(self, controller):
+        # constants
+        self.TEMP_SOURCE = 0
+        self.FREQ_SOURCE = 1
+        self.left_margin = 0
+        self.top_margin = 0
 
+        # main control
         self.controller = controller
-        self.hline = urwid.AttrWrap(urwid.SolidFill(u'_'), 'line')
+        self.main_window_w = []
 
+        # general urwid items
         clock_text = seconds_to_text(self.controller.stress_time)
         self.clock_view = urwid.Text(('bold text', clock_text), align="center")
-
-        self.mode_buttons = []
         self.refresh_rate_ctrl = urwid.Edit(('bold text', u'Refresh[s]:'),
                                             self.controller.refresh_rate)
+        self.hline = urwid.AttrWrap(urwid.SolidFill(u'_'), 'line')
+
+        self.mode_buttons = []
 
         # Visible graphs are the graphs currently displayed, this is a
         # subset of the available graphs for display
         self.visible_graphs = {}
+        self.graphs = {}
+        self.available_graphs = {}
         self.graph_place_holder = urwid.WidgetPlaceholder(urwid.Pile([]))
+        self.summaries = []
+        self.available_summaries = []
 
-        self.main_window_w = []
-
-        # construct temprature graph and source
-        self.temp_source = TempSource(self.controller.custom_temp,
-                                      self.controller.temp_thresh)
-        self.freq_source = FreqSource(is_admin)
+        # construct temperature graph and source
+        self.source_list = []
+        self.source_list.append(TempSource(self.controller.custom_temp,
+                                           self.controller.temp_thresh))
+        self.source_list.append(FreqSource(is_admin))
 
+        # construct the variouse menus during init phase
         self.stress_menu = StressMenu(self.on_menu_close)
         self.help_menu = HelpMenu(self.on_menu_close)
         self.about_menu = AboutMenu(self.on_menu_close)
-        self.sensors_menu = SensorsMenu(self.on_sensors_menu_close, self.temp_source, self.freq_source)
+        self.sensors_menu = \
```

Comment:
Better to just add a new line after the `,`

Reply:
fixed